app/src/main/java/fr/free/nrw/commons/nearby/NearbyPlaces.java

```diff
@@ -9,23 +10,156 @@
 import java.io.IOException;
 import java.io.InputStreamReader;
 import java.net.URL;
+import java.net.URLConnection;
+import java.net.URLEncoder;
 import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 
 public class NearbyPlaces {
 
     private static final String TAG = NearbyPlaces.class.getName();
-    static List<Place> places = null;
+    private static final int MIN_RESULTS = 20;
+    private static final int MAX_RESULTS = 500;
+    private static final double RADIUS_INITIAL = 0.5;
+    private static final double RADIUS_MULTIPLIER = 3.0;
+    private static final String WIKIDATA_QUERY_URL = "https://query.wikidata.org/sparql?query={{QUERY}}";
+    private static final String WIKIDATA_QUERY_TEMPLATE =
```

Comment:
I realize this is ugly (and produces a lot of code style warnings) but couldn't think of a quick alternative.